### PR TITLE
chore: label-actions comments when logs:problem label is applied

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- `label-actions` will comment when `logs:problem` label is applied
- Explain `logs:problem` label usage in the `development/issue-labeling.md` docs

## Context:

Originally this PR was just to add a `logs:formatting` label and comment.

We have decided to make a `logs:problem` label and capture all 3 things that can go wrong with the logs:

- No logs at all
- Logs insufficient
- Logs badly formatted (wall of text in issue or comment)

After this PR gets merged, @rarkins will need to create the new label, and add a description to the label.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Do you think we need a separate label/text for each type of logs problem, or can we combine? I was thinking we can maybe combine these three cases:
- Logs are missing and we need them (including instructions on how to log locally or retrieve from the app)
- Logs provided are not sufficient and user needs to provide more
- User should format them

Or maybe we now have so many labels.. who cares if we split them?

**Comment by HonkingGoose** [comment]
So we have 2 directions we can take this.
I have listed the options below for discussion.

## Option 1: separate `logs` labels

- `logs:needed`: Explain why we need a log, and explain how to log locally/retrieve from app + how to format.
- `logs:insufficient`: Explain that provided logs are not sufficient, explain how to log locally/retrieve from app + how to format.
- `logs:formatting`: Explain how to format provided logs.

Having 3 `logs` type labels might be too much, as we already have a lot of labels.

## Option 2: one `logs:problem` label to rule them all

`logs:problem`: how to log locally/retrieve from app + how to format logs

```markdown
There's a problem with the logs, we either need a log, or better logs, or the current logs are formatted poorly.
Depending on what situation applies follow one, some or all instructions:

Do .... to retrieve logs from app.
Do .... to retrieve logs from local instance.
Do .... to format the logs properly.
```

I like the simplicity of just one catch-all label, if there's something wrong, just slap a `logs:problem` label on it and be done with it. 😄 
We do need to take care to craft a comment that works for all 3 use-cases.

**Comment by rarkins** [comment]
Option 2!

**Comment by HonkingGoose** [comment]
Well I made a bunch of changes:

- Add 3 sections (No logs at all, Insufficient logs, Formatting your logs)
- Provide instructions for hosted app, and self-hosted (DRAFT!)
- Use backticks in formatting your logs example (now that I use 4 spaces for indentation backticks work properly again... 🎉 )

Ready for another round of review and suggestions.

I you want to see the current output of the `draft v2` commit go to: 
https://github.com/HonkingGoose/test-label-action-markdown-log-hint/issues/8

**Comment by rarkins** [comment]
Is it possible for this comment to include details/ summary collapsed sections of its own? Eg one for each case

**Comment by HonkingGoose** [comment]
> Is it possible for this comment to include details/ summary collapsed sections of its own? Eg one for each case

That's a great idea! 😄 And we can actually do this, though the formatting is getting really tricky now. 👻 

See https://github.com/HonkingGoose/test-label-action-markdown-log-hint/issues/10 for a example of what the draft v3 commit gives you.

We still need to improve the text for finding logs on self-hosted instances of Renovate, as that is still placeholder text basically.

File: .github/label-actions.yml [hunk 1]
```diff
@@ -27,3 +27,75 @@
...
+
+    If there's no log posted yet, we need you to find and copy/paste the log into the issue template.
+
+
+    #### Finding logs on hosted app
+
+    <details><summary>Click me to read instructions</summary>
+
+
+    If you use the Renovate app (GitHub/GitLab/etc):
```
Comment (HonkingGoose) [inline .github/label-actions.yml:50]:
```suggestion
    If you use the Renovate app (GitHub):
```

Maybe we should strip GitLab, as that is currently only self-hosted?

Comment (rarkins) [reply .github/label-actions.yml:50]: Agreed

File: .github/label-actions.yml [hunk 2]
```diff
@@ -27,3 +27,75 @@
...
+    1. You are now in the correct repository log overview screen
+
+    1. Copy/paste the correct log
+
+    1. Follow the steps in the **formatting your logs** section
+
+    </details>
+
+
+    #### Finding logs on self-hosted app
```
Comment (rarkins) [inline .github/label-actions.yml:65]:
```suggestion
    #### Finding logs when self-hosting
```

File: .github/label-actions.yml [hunk 3]
```diff
@@ -27,3 +27,75 @@
...
+
+    <details><summary>Click me to read instructions</summary>
+
+
+    If you're running self-hosted, run with `--log-level=debug` or LOG_LEVEL=debug and search for whatever dependency/branch/PR that is causing the problem.
```
Comment (rarkins) [inline .github/label-actions.yml:70]:
```suggestion
    If you're running self-hosted, run with `LOG_LEVEL=debug` in your environment variables and search for whatever dependency/branch/PR that is causing the problem.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: .github/label-actions.yml [hunk 4]
```diff
@@ -27,3 +27,75 @@
...
+    <details><summary>Click me to read instructions</summary>
+
+
+    Please put your logs in a `<details>` and `<summary>` element like this:
+
+
+        <details><summary>Click me to see logs</summary>
+
+        ```
+        Copy/paste your log here
```
Comment (HonkingGoose) [inline .github/label-actions.yml:96]:
```suggestion
        Copy/paste any log here, between the starting and ending backticks
```

**Comment by HonkingGoose** [comment]
I think this is ready for another review, and possibly a merge. 😄 

See https://github.com/HonkingGoose/test-label-action-markdown-log-hint/issues/11 for a live preview of the message from commit 156c060.

**Comment by HonkingGoose** [comment]
Do you want me to update the `development/issue-labeling.md` docs within this PR as well?

I think it makes sense to add the label and some explanatory text in the [issue-labeling, Housekeeping](https://github.com/renovatebot/renovate/blob/master/docs/development/issue-labeling.md#housekeeping) section.

**Comment by rarkins** [comment]
Great job with figuring out the nested formatting!

File: docs/development/issue-labeling.md [hunk 5]
```diff
@@ -150,6 +151,12 @@ This label is sometimes picked up by tools or websites that try to encourage peo
 
 Add the label `help wanted` to indicate that we need the original poster or someone else to do some work or it is unlikely to get done.
 
+Add a label `logs:problem` to indicate that there's a problem with the logs, and the contributor needs to do one of these things:
+
+1. Provide logs (if there are none yet)
+1. Provide more logs (in case current log insufficient)
```
Comment (HonkingGoose) [inline docs/development/issue-labeling.md:157]:
```suggestion
1. Provide more logs (in case current logs are insufficient)
```

**Review by rarkins — APPROVED** [review]

**Comment by HonkingGoose** [comment]
> Great job with figuring out the nested formatting!

Thank you! 😄 

And thank you again for merging my work! ✨ 

Can you create the label `logs:problem`, I don't have the rights needed to create labels here, I can only apply existing labels. 😉

**Comment by HonkingGoose** [comment]
@viceice can you please create a new label `logs:problem`? I don't have the rights to create new labels. 😉

**Comment by HonkingGoose** [comment]
@viceice or @rarkins Thank you for making the label! 🎉 

By the way the label action only works on "issues" not on "pull requests":

https://github.com/renovatebot/renovate/blob/ef447c7ab9ef0b4ea47469caf58a72bae15021ce/.github/workflows/label-actions.yml#L3-L5

I'll go try it out on a issue of mine. 😄

**Pull request merged**